### PR TITLE
Add Live Data to Reviews Panel

### LINKED ISSUE #1036: Activity Panel: Add Live Data to Reviews Panel
Child task of #199

Currently the Reviews Activity Panel is loading a placeholder/fake product review. This task involves adding live data to the panel, akin to the work that took place in #969 for the Orders Panel.

### PR DESCRIPTION
**Pull request by emdashcodes**
Fixes #1036.

This PR hooks up the 'reviews' activity panel to live data.

It also modifies the `/products/*` and `/products/reviews/*` endpoints, by allowing embedding (https://developer.wordpress.org/rest-api/using-the-rest-api/linking-and-embedding/) of some product fields when `?_embed=1` is present in the query string.

This allows us to get info like the product name, permalink, and image without extra requests. It also **matches the behavior** of WP-API Core endpoints. For example on a comment, you can receive some posts field. So this change makes things easier for us, and makes sense from a consistency point of view. (You could already embed some user info via the reviews endpoint).

### Screenshots

screen shot 2018-12-10 at 1 25 49 pm

### Detailed test instructions:

- Run `phpunit` and make sure all tests pass
- Click reviews on the activity panel
- Verify that your reviews properly load in

### CODE REVIEW

File: client/header/activity-panel/panels/reviews.js [hunk 1]
```diff
@@ -22,62 +23,18 @@ import {
...
-	}
-
-	componentWillUnmount() {
-		clearTimeout( this.interval );
-	}
-
 	renderReview( review ) {
-		const product = demoProducts[ 0 ];
-
+		const product = review._embedded.up[ 0 ];
```
Comment (joshuatf) [inline client/header/activity-panel/panels/reviews.js]: Probably overly cautious, but do we want to check that these properties exist before rendering?

**Review by joshuatf — CHANGES_REQUESTED** [review]
Looking good and testing well!  Nice work on this @justinshreve 👍 

I've added a couple comments where I think we can reduce a bit of this code, but otherwise things are looking good to merge.

2 side notes that may be better suited for separate PRs:
1. Should we add testing for the API response?
2. Products without an image that use the placeholder image look like misaligned images (since the background is white).  Do we need to make any changes here or do they look good to as is @LevinMedia ?

screen shot 2018-12-11 at 3 07 14 pm

File: includes/api/class-wc-admin-rest-products-controller.php [hunk 2]
```diff
@@ -0,0 +1,642 @@
...
+	 * @var string
+	 */
+	protected $namespace = 'wc/v3';
+
+	/**
+	 * Get the Product's schema, conforming to JSON Schema.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
```
[the remark this answers is not in the document]
Comment (justinshreve) [reply includes/api/class-wc-admin-rest-products-controller.php]: The difference was adding `embed` to the context field. But good call, I've simplified this and reduced a lot of lines.

**Review by joshuatf — APPROVED** [review]

**Comment by joshuatf** [comment]
@justinshreve Woo- all that red!  Looks good and new tests are all passing.

I think we can address the placeholder image in a separate issue ( #1067 ) since it's a bit outside the scope of this PR.

LGTM!

**Comment by justinshreve** [comment]
Thank you for logging that!

**Pull request merged**